ehr/test/src/org/labkey/test/tests/ehr/AbstractGenericEHRTest.java

```diff
@@ -300,6 +306,160 @@ private void testUserAgainstAllStates(@LoggedParam EHRUser user)
         resetErrors();  //note: inserting records without permission will log errors by design.  the UI should prevent this from happening, so we want to be aware if it does occur
     }
 
+    protected List<String> skipLinksForValidation()
+    {
+        return List.of(
+                "showAllErrors.view",
+                "query-exportRowsExcel.view",
+                "ldk-runNotification.view"  // need to scope notifications to enabled modules then can remove this
+        ); // Override if there are links to pages that are known to throw errors
+    }
+
+    protected List<String> skipLinksForCrawling()
+    {
+        return List.of(
+                "project-begin.view",
+                "query-begin.view",
+                "query-searchPanel.view",
+                "query-executeQuery.view",
+                "study-manageStudy.view",
+                "ehr-animalHistory.view",
+                "ehr-updateQuery.view",
+                "ehr-updateTable.view",
+                "ehr-populateLookupData.view",
+                "ehr-ensureQcStates.view",
+                "ehr-ehrTemplates.view",
+                "ehr-primeDataEntryCache.view",
+                "ehr-cacheLivingAnimals.view",
+                "core-modulePropertyAdmin.view",
+                "dataintegration-begin.view",
+                "ldk-updateQuery",
+                "junit-begin.view"
+        );
+    }
+
+    private String validLink(WebElement anchor)
+    {
+        boolean clickable = false;
+        String validUrl = null;
+
+        String href = anchor.getDomAttribute("href");
+        if (href != null && !href.startsWith("#"))
+        {
+            String decodedHref = EscapeUtil.decodeUriPath(href);
+            if (skipLinksForValidation().stream().anyMatch(s -> decodedHref.toLowerCase().contains(s.toLowerCase())))
+            {
+                log(href + " is specified as an exception to link validation. Skipping validation.");
+                return validUrl;
```

Comment:
`validUrl` will always be `null` here. Returning `validUrl` adds unnecessary cognitive load to anybody reading this method.
```suggestion
                return null;
```